### DISCUSSION TITLE
[Feature Request]: Macro Deck support !

### DISCUSSION THREAD

**Discussion by johnvo0918** [pear-devs/pear-desktop#2329] [General]
Add support to macro deck so I can reuse my old devices for media control !

**Comment by ArjixWasTaken** [comment]
Looks like macro deck has its own system for plugins, and plugins are C# dll files.
That would be an entirely different project, and cannot be part of our client.

Macro deck has [some documentation](https://github.com/Macro-Deck-App/Macro-Deck/wiki/Plugin-API) on making a plugin, and we don't need to do anything on our part for you (or someone else) to make a plugin.

One can make a macro deck plugin that calls our app protocol to do certain actions.

https://github.com/th-ch/youtube-music/blob/master/src/providers/song-controls.ts#L27-L80

e.g. visiting `youtubemusic://like` on your browser, or executing `start youtubemusic://like` from cmd
